src/components/qrModal/qrModalView.tsx

```diff
@@ -97,42 +114,145 @@ export const QRModal = ({}: QRModalProps) => {
 
   const onSuccess = (e) => {
     setIsScannerActive(false);
-    _handleDeepLink(e.data);
+    if (isHiveUri(e.data)) {
+      _handleHiveUri(e.data);
+    } else {
+      _handleDeepLink(e.data);
+    }
   };
 
-  const _handleDeepLink = async (url) => {
-    setIsProcessing(true);
-    const deepLinkData = await deepLinkParser(url, currentAccount);
-    const { name, params, key } = deepLinkData || {};
-    setIsProcessing(false);
-    if (name && params && key) {
+  const _handleHiveUri = async (uri: string) => {
+    try {
       setIsScannerActive(false);
       _onClose();
-      RootNavigation.navigate(deepLinkData);
-    } else {
-      Alert.alert(
-        intl.formatMessage({ id: 'qr.unsupported_alert_title' }),
-        intl.formatMessage({ id: 'qr.unsupported_alert_desc' }),
-        [
+      if (!isLoggedIn) {
+        showLoginAlert({ intl });
+        return;
+      }
+      if (isPinCodeOpen) {
+        RootNavigation.navigate({
+          name: ROUTES.SCREENS.PINCODE,
+          params: {
+            callback: () => _handleHiveUriTransaction(uri),
+          },
+        });
+      } else {
+        _handleHiveUriTransaction(uri);
+      }
+    } catch (err) {
+      _showInvalidAlert();
+    }
+  };
+
+  const _handleHiveUriTransaction = async (uri: string) => {
+    if (get(currentAccount, 'local.authType') === authType.STEEM_CONNECT) {
+      await delay(500); // NOTE: it's required to avoid modal mis fire
+      dispatch(
+        showWebViewModal({
+          uri: uri,
+        }),
+      );
+      return;
+    }
+
+    const parsed = hiveuri.decode(uri);
+    // resolve the decoded tx and params to a signable tx
+    let { tx, signer } = hiveuri.resolveTransaction(parsed.tx, parsed.params, {
```

Comment:
Might be a duplication, resolve transaction also happening inside `handleHiveUriOperation` in dhive.js

Reply:
@noumantahir this is not duplication, both server the different purpose, above resolveTransaction method is being used to get the tranaction details and show in action modal, 2nd one is being used in dhive.js for signing transaction